acceptance/tests/lookup/config5_interpolation.rb

```diff
@@ -0,0 +1,119 @@
+test_name 'C99578: hiera5 lookup config with interpolated scoped nested variables' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type + '1')
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+
+  step "create environment hiera5.yaml and environment data" do
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hiera.yaml", <<-HIERA)
+---
+version: 5
+defaults:
+  datadir: 'hieradata'
+  data_hash: yaml_data
+hierarchy:
+  - name: "Global settings"
+    path: "global.yaml"
+  - name: "Role specific settings"
+    paths:
+      - "roles/%{::roles.0}.yaml"
+  - name: "Other Role specific settings"
+    paths:
+      - "roles/%{roles2.0}.yaml"
+  - name: "scoped variable"
+    paths:
+      - "roles/%{::myclass::myvar.0}.yaml"
+  - name: "nested hash variable"
+    paths:
+      - "roles/%{::hash_array.key1.0}.yaml"
+    HIERA
+
+    on(master, "mkdir -p #{fq_tmp_environmentpath}/hieradata/roles")
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/global.yaml", <<-YAML)
+roles:
+  - test1
+roles2:
+  - test2
+data:
+  - "from global"
+    YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/roles/test1.yaml", <<-YAML)
+data:
+  - 'from test1'
+    YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/roles/test2.yaml", <<-YAML)
+data:
+  - 'from test2'
+    YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/roles/test3.yaml", <<-YAML)
+data:
+  - 'from test3'
+    YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/roles/test4.yaml", <<-YAML)
+data:
+  - 'from test4'
+    YAML
+
+    create_sitepp(master, tmp_environment, <<-SITE)
+class myclass {
+  $myvar = ['test3']
+}
+include myclass
+
+$hash_array = {key1 => ['test4']}
+
+$roles = lookup('roles')
+$data = lookup('data', Array[String], 'unique')
+notify{"data: ${data}":}
+$hiera_array_data = hiera_array('data')
+notify{"hiera_array_data: ${hiera_array_data}":}
+
+$roles2 = lookup('roles2')
+$data2 = lookup('data', Array[String], 'unique')
+notify{"data2: ${data2}":}
+$hiera_array_data2 = hiera_array('data')
+notify{"hiera_array_data2: ${hiera_array_data2}":}
+    SITE
+
+    on(master, "chmod -R 775 #{fq_tmp_environmentpath}")
+  end
+
+  with_puppet_running_on(master,{}) do
```

Comment:
I get errors when running this test on Ubuntu 14.04 (it installs Ruby 1.9.3). It passes when using Ubuntu 16.10.
```
Permission denied - /etc/puppetlabs/puppet/hiera.yaml at /etc/puppetlabs/code/environments/config5_interpolation1_u3jifxre/manifests/site.pp:8:10
```